sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_upload_helper.py

```diff
@@ -0,0 +1,82 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+# pylint: disable=no-self-use
+
+from ._deserialize import (
+    process_storage_error)
+from ._generated.models import (
+    StorageErrorException,
+)
+from ._shared.response_handlers import return_response_headers
+from ._shared.uploads import (
+    upload_data_chunks,
+    DataLakeFileChunkUploader)
+
+
+def _any_conditions(modified_access_conditions=None, **kwargs):  # pylint: disable=unused-argument
+    return any([
+        modified_access_conditions.if_modified_since,
+        modified_access_conditions.if_unmodified_since,
+        modified_access_conditions.if_none_match,
+        modified_access_conditions.if_match
+    ])
+
+
+def upload_datalake_file(  # pylint: disable=unused-argument
+        client=None,
+        stream=None,
+        length=None,
+        overwrite=None,
+        validate_content=None,
+        max_concurrency=None,
+        **kwargs):
+    try:
+        if length == 0:
+            return {}
+        properties = kwargs.pop('properties', None)
+        path_http_headers = kwargs.pop('path_http_headers', None)
+        modified_access_conditions = kwargs.pop('modified_access_conditions', None)
+
+        if not overwrite:
+            # if customers didn't specify access conditions, they cannot flush data to existing file
+            if not _any_conditions(modified_access_conditions):
+                modified_access_conditions.if_none_match = '*'
+            if properties:
+                raise ValueError("metadata can be set only when overwrite is enabled")
```

Comment:
Does this mean that if `overwrite=False` - we assume the file already exists?

Reply:
Yes, we assumes an empty file is there. 
We thought about enable use to upload data to any file, while not everyone likes that, the reason is the function shouldn't take too many responsibilities.